tealer/teal/instructions/transaction_field.py

```diff
@@ -24,6 +25,13 @@ def version(self) -> int:
         """Teal version this field is introduced in and supported from."""
         return self._version
 
+    def __eq__(self, other: Any) -> bool:
+        # pylint: disable=unidiomatic-typecheck
+        return type(other) == type(self)
+
+    def __hash__(self) -> int:
+        return hash(str(self))
+
```

Comment:
^^ Won't this approach result in errors when transaction fields from `gtxn` instruction are tracked alongside `txn` instruction?

1. txn RekeyTo
2. gtxn 3 RekeyTo

RekeyTo from `1` and `2` would have same `str(self)` and `type`. They will be considered equal during stack value analysis. `txn RekeyTo` and `gtxn 3 RekeyTo` does not necessarily be equal and most of the times will not be equal.

I think we need a better way to compare and hash TransactionField objects.